lib/activerecord-bitemporal/bitemporal.rb

```diff
@@ -131,9 +131,12 @@ def with_bitemporal_option(**opt)
 
         def find(*ids)
           return super if block_given?
-          expects_array = ids.first.kind_of?(Array) || ids.size > 1
-          ids = ids.first if ids.first.kind_of?(Array)
-          where(bitemporal_id_key => ids).yield_self { |it| expects_array ? it&.to_a : it&.take }.presence || raise(::ActiveRecord::RecordNotFound)
+          all.spawn.yield_self { |obj|
+            def obj.primary_key
+              "bitemporal_id"
```

Comment:
It looks strange, but I understand that it doesn't have side effects because `obj` is cloned.